src/NHSDPublicBrowseAcceptanceTests.Actions/Pages/ViewASolution.cs

```diff
@@ -12,9 +14,10 @@ public ViewASolution(IWebDriver driver) : base(driver)
         {
         }
 
-        public void PageDisplayed()
+        public void PageDisplayed(string url)
         {
-            wait.Until(s => s.Url.Contains("/view-solution/"));
+            // Should be a better way to do this that doesn't rely on RegEx matching
+            Regex.Match(driver.Url, $@"{url}/solutions/(foundation|all)/(.*)").Success.Should().BeTrue();            
```

Comment:
isn't /(.*) a capture group?
what about just /* ?

Reply:
A SolutionID is in the URL, but as we can't always guarantee the format for that, I've set it to just capture everything after the (foundation|all) regex capture to bypass this sticking point

Reply:
Updated in next commit